azure-pipelines/createBuildDirectories.sh

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env bash
+
+# This script creates the necessary directories that are required 
+# for the the linux dotnet builds to work. Currently, there is an 
+# issue in the linux dotnet core where it does not consider 
+# the letter casing of the localization directories it creates. It 
+# is not an issues in the window version because the dirs are case 
+# insensitive.
+# The issue that tracks this bug is https://github.com/dotnet/msbuild/issues/5768
+
+# The script need to run from the repo root
+
+net6projectArray=(
+    "./src/Microsoft.InsightsGenerator"
+    "./src/Microsoft.Kusto.ServiceLayer"
+    "./src/Microsoft.SqlTools.Credentials"
+    "./src/Microsoft.SqlTools.Hosting"
+    "./src/Microsoft.SqlTools.ResourceProvider"
+    "./src/Microsoft.SqlTools.ResourceProvider.Core"
+    "./src/Microsoft.SqlTools.ResourceProvider.DefaultImpl"
+    "./src/Microsoft.SqlTools.ServiceLayer"
+    "./test/Microsoft.InsightsGenerator.UnitTests"
+    "./test/Microsoft.Kusto.ServiceLayer.UnitTests"
+    "./test/Microsoft.SqlTools.ManagedBatchParser.IntegrationTests"
+    "./test/Microsoft.SqlTools.ServiceLayer.IntegrationTests"
+    "./test/Microsoft.SqlTools.ServiceLayer.PerfTests"
+    "./test/Microsoft.SqlTools.ServiceLayer.Test.Common"
+    "./test/Microsoft.SqlTools.ServiceLayer.TestDriver"
+    "./test/Microsoft.SqlTools.ServiceLayer.TestDriver.Tests"
+    "./test/Microsoft.SqlTools.ServiceLayer.TestEnvConfig"
+    "./test/Microsoft.SqlTools.ServiceLayer.UnitTests"
+    "./test/Microsoft.SqlTools.Test.CompletionExtension"
+)
+
+netStandard2ProjectArray=(
+    "./src/Microsoft.SqlTools.ManagedBatchParser"
+)
+
+# Please update the framework vars when updating target framework for the projects
+framework6="/bin/Debug/net6.0/"
+framework2="/bin/Debug/netstandard2.1/"
+
+requiredLocDirectories=(
+    "cs"
+    "cs-CZ"
+    "de"
+    "de-DE"
+    "es"
+    "es-ES"
+    "fr"
+    "fr-FR"
+    "hu-HU"
+    "it"
+    "it-IT"
+    "ja"
+    "ja-JP"
+    "ko"
+    "ko-KR"
+    "nl-NL"
+    "pl"
+    "pl-PL"
+    "pt-br"
```

Comment:
We actually need to make all these folders? It seems odd that we need to create the folders for both cases, shouldn't just one be enough?

Reply:
Removed a few directories that are unnecessary.

Reply:
there are 2 issues:
1. dotnet build not able to create directories - we are adding a workaround in this PR
2. Linux is case sensitive, but we have mixed casing in multiple places: our own code and the nuget packages. in order to have just one folder per language, we have to make sure the loc folders in all nuget packages are following the same casing convention. 

that is what I was saying this PR is not ideal, but acceptable at least we can have the test running.

Reply:
I'm still confused why we need multiple folders for the same loc - like `ru` and `ru-RU`. And then also still have some mixed case ones like `zh-hans` and `zh-HANS`

@aasimkhan30 Could you put a comment in here explaining why all these different ones are necessary? The comment you have explains why we need to create the folders in the first place, but it doesn't really explain why we have to make "duplicate" folders for the different languages.

Reply:
I can add Alan's explanation to the description

Reply:
Oh gotcha, so it's because of the nuget packages we consume. (because we can control the casing in our own repo so that shouldn't be a problem...)